### PR TITLE
filtering changes required for list methods, misc improvements

### PR DESCRIPTION
**Pull request by gesparza3**
This PR includes changes to implement Grafeas listing queries for the Elasticsearch backend. Something to note is that `assertJsonHasValues` does not support searching for values within arrays so in the unit tests for listing we had to unmarshal the performed request to read the terms within listing queries.

### CODE REVIEW

**Comment by mrparkers** [comment]
> Something to note is that `assertJsonHasValues` does not support searching for values within arrays

Yes it does: https://github.com/liatrio/grafeas-elasticsearch/blob/a0898b8b437eb9e83e288b67016ea76ec5c1cf43/go/v1beta1/storage/elasticsearch_test.go#L831-L838

But using well defined types is probably better anyways

File: go/v1beta1/storage/elasticsearch.go [hunk 1]
```diff
@@ -198,9 +216,59 @@ func (es *ElasticsearchStorage) GetProject(ctx context.Context, projectID string
...
+
+	body := &esSearch{}
+
+	log := es.logger.Named("ListProjects")
+
+	if filter != "" {
+		filterQuery, err := filtering.ParseExpressionEntrypoint(filter)
+		if err != nil {
+			// There can be many parse errors in a filter, this returns the first error found
+			return nil, "", createError(log, "error while parsing filter", errors.New(err[0].Message)) //library to consolidate array of errs
```
Comment (aalsabag) [inline go/v1beta1/storage/elasticsearch.go:229]: https://liatrio.atlassian.net/browse/RODE-39
we'll come back to this

File: go/v1beta1/storage/elasticsearch.go [hunk 2]
```diff
@@ -488,28 +564,83 @@ func (es *ElasticsearchStorage) DeleteOccurrence(ctx context.Context, pID, oID s
 
 // GetNote returns the note with project (pID) and note ID (nID)
 func (es *ElasticsearchStorage) GetNote(ctx context.Context, pID, nID string) (*pb.Note, error) {
-	return nil, nil
+	log := es.logger.Named("GetNote")
```
Comment (aalsabag) [inline go/v1beta1/storage/elasticsearch.go:567]: Asking @mrparkers to check this to make sure it matches his getOccurrence (i think that's what he was working on)

File: go/v1beta1/storage/elasticsearch.go [hunk 3]
```diff
@@ -488,28 +564,83 @@ func (es *ElasticsearchStorage) DeleteOccurrence(ctx context.Context, pID, oID s
...
 	var (
-		os   []*pb.Occurrence
-		body strings.Builder
+		notes []*pb.Note
 	)
+	body := &esSearch{}
+	if filter != "" {
+		filterQuery, err := filtering.ParseExpressionEntrypoint(filter)
+		stringVal, _ := json.Marshal(filterQuery)
+		fmt.Println(string(stringVal))
```
Comment (aalsabag) [inline go/v1beta1/storage/elasticsearch.go:626]: remove

**Review by aalsabag — APPROVED** [review]

File: go/v1beta1/storage/elasticsearch.go [hunk 4]
```diff
@@ -159,9 +169,17 @@ func (es *ElasticsearchStorage) GetProject(ctx context.Context, projectID string
 	projectName := fmt.Sprintf("projects/%s", projectID)
 	log := es.logger.Named("GetProject").With(zap.String("project", projectName))
 
-	searchTerm, err := createElasticsearchSearchTermQuery(map[string]interface{}{
-		"name": projectName,
-	})
+	filterQuery, errs := filtering.ParseExpressionEntrypoint(fmt.Sprintf(`name=="%s"`, projectName))
```
Comment (mrparkers) [inline go/v1beta1/storage/elasticsearch.go:172]: We don't really need to do this if the filter is already known beforehand, right?  It's always going to be the same search query.  We should update this to search in the same way that `CreateProject` does.

File: go/v1beta1/storage/elasticsearch.go [hunk 5]
```diff
@@ -209,9 +277,17 @@ func (es *ElasticsearchStorage) DeleteProject(ctx context.Context, projectID str
 	log := es.logger.Named("DeleteProject").With(zap.String("project", projectName))
 	log.Info("deleting project")
 
-	searchTerm, err := createElasticsearchSearchTermQuery(map[string]interface{}{
-		"name": projectName,
-	})
+	filterQuery, errs := filtering.ParseExpressionEntrypoint(fmt.Sprintf(`name=="%s"`, projectName))
```
Comment (mrparkers) [inline go/v1beta1/storage/elasticsearch.go:280]: Same comment here

File: go/v1beta1/storage/elasticsearch.go (hunk 2 above)
Comment (mrparkers) [reply go/v1beta1/storage/elasticsearch.go:567]: It's similar enough.  I'll update some things after this is merged to be more consistent with the new `GetOccurrence` implementation.

File: go/v1beta1/storage/elasticsearch.go [hunk 6]
```diff
@@ -488,28 +564,83 @@ func (es *ElasticsearchStorage) DeleteOccurrence(ctx context.Context, pID, oID s
...
+	return note, nil
 }
 
 // ListNotes returns up to pageSize number of notes for this project (pID) beginning
 // at pageToken (or from start if pageToken is the empty string).
-func (es *ElasticsearchStorage) ListNotes(ctx context.Context, pID, filter, pageToken string, pageSize int32) ([]*pb.Note, string, error) {
+func (es *ElasticsearchStorage) ListNotes(ctx context.Context, projectID, filter, pageToken string, pageSize int32) ([]*pb.Note, string, error) {
 	log := es.logger.Named("ListNotes")
-	log.Debug("Project ID", zap.String("pID", pID))
+	log.Debug("Project ID", zap.String("projectID", projectID))
```
Comment (mrparkers) [inline go/v1beta1/storage/elasticsearch.go:617]: I'd rather add this as a field to the current logger rather than log this on its own

**Pull request merged**